types/v56/types.go

```diff
@@ -445,49 +445,96 @@ type AdminVdc struct {
 	UniversalNetworkPoolReference *Reference     `xml:"UniversalNetworkPoolReference,omitempty"` // Reference to a universal network pool
 }
 
-// VdcStorageProfile represents the parameters to create a storage profile in an organization vDC.
+// CreateVdcStorageProfile represents the parameters to create a storage profile in an organization vDC.
 // Type: VdcStorageProfileParamsType
 // Namespace: http://www.vmware.com/vcloud/v1.5
 // Description: Represents the parameters to create a storage profile in an organization vDC.
 // Since: 5.1
 // https://code.vmware.com/apis/220/vcloud#/doc/doc/types/VdcStorageProfileParamsType.html
-type VdcStorageProfile struct {
+type CreateVdcStorageProfile struct {
 	Enabled                   bool       `xml:"Enabled,omitempty"`
 	Units                     string     `xml:"Units"`
 	Limit                     int64      `xml:"Limit"`
 	Default                   bool       `xml:"Default"`
 	ProviderVdcStorageProfile *Reference `xml:"ProviderVdcStorageProfile"`
 }
 
+// VdcStorageProfile represents the parameters for fetched storage profile
+// Type: VdcStorageProfileParamsType
+// Namespace: http://www.vmware.com/vcloud/v1.5
+// https://vdc-repo.vmware.com/vmwb-repository/dcr-public/7a028e78-bd37-4a6a-8298-9c26c7eeb9aa/09142237-dd46-4dee-8326-e07212fb63a8/doc/doc/types/VdcStorageProfileType.html
+// https://vdc-repo.vmware.com/vmwb-repository/dcr-public/71e12563-bc11-4d64-821d-92d30f8fcfa1/7424bf8e-aec2-44ad-be7d-b98feda7bae0/doc/doc/types/AdminVdcStorageProfileType.html
+type VdcStorageProfile struct {
+	Xmlns                     string                         `xml:"xmlns,attr"`
+	Name                      string                         `xml:"name,attr"`
+	Enabled                   bool                           `xml:"Enabled,omitempty"`
+	Units                     string                         `xml:"Units"`
+	Limit                     int64                          `xml:"Limit"`
+	Default                   bool                           `xml:"Default"`
+	IopsSettings              *VdcStorageProfileIopsSettings `xml:"IopsSettingsint64"`
+	StorageUsedMB             int64                          `xml:"StorageUsedMB"`
+	IopsAllocated             int64                          `xml:"IopsAllocated"`
```

Comment:
Hard to understand the diff in GitHub, but do I get right, that in reality we have added these three new fields to the old `VdcStorageProfile` struct?

Reply:
Answer in below comment